test/integration/discovery.go

```diff
@@ -269,6 +269,56 @@ var _ = Describe("Discover Cluster", func() {
 		})
 	})
 
+	Describe("discovering pod with pv having affinity rules", func() {
+		It("discovering with pv with affinity rules", func() {
+			//Add Storage Class
+			newStorage, err := createStorageClass(kubeClient)
+			if newStorage == nil && err != nil {
+				framework.Failf("Failed to create Storage Class %s with the error %s", newStorage, err)
+			}
+			_, err = createPV(kubeClient, namespace, newStorage.Name)
+			pvc, err := createVolumeClaim(kubeClient, namespace, newStorage.Name)
+			if pvc == nil && err != nil {
+				framework.Failf("Failed to create PVC %s with the error %s", pvc, err)
+			}
+			dep, err := createDeployResource(kubeClient, depSingleContainerWithResources(namespace, pvc.Name, 1, true, false, false, ""))
+			framework.ExpectNoError(err, "Error creating test resources")
+
+			pod, err := getPodWithNamePrefix(kubeClient, dep.Name, namespace, "")
+			framework.ExpectNoError(err, "Error getting deployments pod")
+			// This should not happen. We should ideally get a pod.
+			if pod == nil {
+				framework.Failf("Failed to find a pod for deployment: %s", dep.Name)
+			}
+
+			var bCommodityRegistered bool
+			var podEntityDTO *proto.EntityDTO
+			entityDTOs, _, err := discoveryClient.DiscoverWithNewFramework("discovery-integration-test")
+			framework.ExpectNoError(err, "Failed completing discovery of test cluster")
+			for _, entityDTO := range entityDTOs {
+				if *entityDTO.EntityType == proto.EntityDTO_CONTAINER_POD &&
+					strings.Contains(*entityDTO.DisplayName, pod.Name) {
+					podEntityDTO = entityDTO
```

Comment:
You can break here once the pods DTO is found

Reply:
Updated the code.